### PR TITLE
Add gas to tx

### PR DESCRIPTION
**Pull request by sunboshan**
Part of https://github.com/ArcBlock/forge/issues/851

### CODE REVIEW

File: lib/protobuf/state.proto [hunk 1]
```diff
@@ -85,8 +86,7 @@ message ForgeState {
   StakeConfig stake_config = 10;
   PokeConfig poke_config = 11;
   repeated CoreProtocol protocols = 12;
-
-  // 12-13 is reserved for future
+  map<string, int32> gas = 13;
```
Comment (tyrchen) [inline lib/protobuf/state.proto]: uint32

File: lib/protobuf/gen/state.pb.ex [hunk 2]
```diff
@@ -199,6 +205,20 @@ defmodule ForgeAbi.ForgeState.StakeSummaryEntry do
   field :value, 2, type: ForgeAbi.StakeSummary
 end
 
+defmodule ForgeAbi.ForgeState.GasEntry do
```
Comment (tyrchen) [inline lib/protobuf/gen/state.pb.ex]: what is this? looks like you're using the map<> but the generated code still has this?

Comment (sunboshan) [reply lib/protobuf/gen/state.pb.ex]: Yes this is generated by the map<>.

File: lib/protobuf/state.proto (hunk 1 above)
Comment (sunboshan) [reply lib/protobuf/state.proto]: fixed

**Pull request merged**